File: libsignal/service/src/main/java/org/whispersystems/signalservice/internal/push/GroupMismatchedDevices.java
package org.whispersystems.signalservice.internal.push;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Represents the body of a 409 response from the service during a sender key send.
 */
public class GroupMismatchedDevices {
  @JsonProperty
  private String uuid;

  @JsonProperty
  private MismatchedDevices devices;

  public GroupMismatchedDevices() {}

  public String getUuid() {
    return uuid;
  }

  public MismatchedDevices getDevices() {
    return devices;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/GroupMismatchedDevices.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.internal.push

/**
 * Represents the body of a 409 response from the service during a sender key send.
 */
data class GroupMismatchedDevices(
  val uuid: String,
  val devices: MismatchedDevices
)
